forc-pkg/src/manifest.rs

```diff
@@ -14,19 +14,19 @@ use sway_utils::constants;
 
 type PatchMap = BTreeMap<String, Dependency>;
 
-/// A [Manifest] that was deserialized from a file at a particular path.
+/// A [PackageManifest] that was deserialized from a file at a particular path.
 #[derive(Clone, Debug)]
-pub struct ManifestFile {
+pub struct PackageManifestFile {
     /// The deserialized `Forc.toml`.
-    manifest: Manifest,
+    manifest: PackageManifest,
```

Comment:
Will there be different files for declaring packages vs declaring a workspace? If they are still within `Forc.toml`, I think I'd find the `PackageManifestFile` naming to be a bit confusing :thinking:

Reply:
Just to clarify the current approach, I can summarize them like the following: 

With this approach package manifests (manifest files that are residing in packages) and workspace manifests are going to be two different files both named `Forc.toml` in different locations in the project structure as they are both manifest files for `forc`. Just like a `Cargo.toml` can be a virtual manifest file (which holds workspace information) at the root of the project structure and package manifest file that holds information about the package itself and resides inside the package's folder.

Since we have two different structures with different functionalities we have `PackageManifest` and `WorkspaceManifest` as different structures. These structures are direct mappings of these two types of manifests. A `PackageManifestFile` and `WorkspaceManifestFile` is a packing of these mappings with their corresponding paths, as throughout the flow of the `forc-pkg` operations we sometimes need to access both the manifest information and the path of that manifest information coming from. 

Since we changed name of the default `Manifest` structure to `PackageManifest` as we now have one more type, I also renamed the structure that holds both the `PackageManifest` and the path for that specific `Forc.toml` file to `PackageManifestFile`. The same logic goes for `WorksapceManifestFile` too. We either need to do it like this or merge them under a single enum and check & produce errors based on their types which was the approach at the beginning but it resulted in a fairly complex code so we decided to separate them as much as possible.

The logic behind the `PackageManifest` wording was the fact that these manifests are going to be inside a package and we are using the "package" name for referencing individual forc projects throughout `forc-pkg`.

These being said I am open for any suggestions too 🙌

Reply:
Ahh okay, got it. Thank you for the detailed explanation!